File: backend/app/schemas/simulation.py
from pydantic import BaseModel

from app.schemas.location import LocationType
from app.schemas.scoring import ColorProbabilities
from app.schemas.session import SunEvent

DISCLAIMER = (
    "This is an AI-generated illustrative representation of expected sky "
    "conditions, not a guaranteed photograph of the future."
)


class SimulationRequest(BaseModel):
    """Fields the client already has from a /session result — no server-side
    session state needed to request a preview for the location the user picked.
    """

    location_name: str
    location_type: LocationType
    event: SunEvent
    cloud_cover_summary: str
    visibility_likelihood: float
    color_probabilities: ColorProbabilities


class SimulationResponse(BaseModel):
    prompt: str
    image_url: str | None
    disclaimer: str = DISCLAIMER
    provider_status: str  # "generated" | "not_configured" | "error"

File: client/lib/models/simulation.dart
import 'location_type.dart';
import 'session_request.dart';
import 'session_response.dart';

class SimulationRequest {
  final String locationName;
  final LocationType locationType;
  final SunEvent event;
  final String cloudCoverSummary;
  final double visibilityLikelihood;
  final ColorProbabilities colorProbabilities;

  const SimulationRequest({
    required this.locationName,
    required this.locationType,
    required this.event,
    required this.cloudCoverSummary,
    required this.visibilityLikelihood,
    required this.colorProbabilities,
  });

  Map<String, dynamic> toJson() => {
        'location_name': locationName,
        'location_type': locationType.apiValue,
        'event': event.apiValue,
        'cloud_cover_summary': cloudCoverSummary,
        'visibility_likelihood': visibilityLikelihood,
        'color_probabilities': colorProbabilities.toJson(),
      };
}

class SimulationResponse {
  final String prompt;
  final String? imageUrl;
  final String disclaimer;
  final String providerStatus;

  const SimulationResponse({
    required this.prompt,
    required this.imageUrl,
    required this.disclaimer,
    required this.providerStatus,
  });

  factory SimulationResponse.fromJson(Map<String, dynamic> json) => SimulationResponse(
        prompt: json['prompt'] as String,
        imageUrl: json['image_url'] as String?,
        disclaimer: json['disclaimer'] as String,
        providerStatus: json['provider_status'] as String,
      );
}
